kubesphere/ks-devops
# Add a webhook to process pipeline event

### What type of PR is this?
/kind feature

### What this PR does / why we need it:

1. Add a webhook to process pipeline event
    1. Create PipelineRun when event type is `run.initialize`
2. Remove the request of synchronizing PipelineRun and disable PipelineRun synchronizer by default

Please refer to 
- https://github.com/kubesphere/ks-devops/discussions/433
- https://github.com/kubesphere/ks-devops/issues/65
### Which issue(s) this PR fixes:

Fixes #325

### Special notes for reviewers:

Docker images for test
```
# DevOps Jenkins
kubespheredev/ks-jenkins:master
# DevOps APIServer
johnniang/devops-apiserver:dev-v3.2.1-54886e1
# DevOps Controller
johnniang/devops-controller:dev-v3.2.1-54886e1
```

Steps to test:

- Replace docker images above for Deployment devops-apiserver and devops-controller
- Create a simple Pipeline in KubeSphere console
- Trigger the corresponding Pipeline in Jenkins
- See the PipelineRuns in KubeSphere console

```go
 W0118 02:18:45.926535       1 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.

 W0118 02:18:45.928388       1 options.go:141] ks-apiserver starts without redis provided, it will use in memory cache. This may cause inconsistencies when running ks-apiserver with multiple replicas.

 I0118 02:18:45.971727       1 sonarqube.go:39] Sonarqube integration is disabled

 I0118 02:18:45.973780       1 sonarqube.go:39] Sonarqube integration is disabled

 [restful] 2022/01/18 02:18:46 log.go:33: [restful/swagger] listing is available at /apidocs.json

 [restful] 2022/01/18 02:18:46 log.go:33: [restful/swagger] /apidocs/ is mapped to folder bin/swagger-ui

 I0118 02:18:46.004237       1 apiserver.go:220] Start cache objects

 I0118 02:18:46.226483       1 apiserver.go:275] Finished caching objects

 I0118 02:18:46.226576       1 apiserver.go:184] Start listening on :9090

 I0118 02:25:25.826024       1 workflow_handler.go:116] Created a PipelineRun: my-devops-projectdq8jj/simple-pipeline-qst5s

 I0118 02:26:02.346993       1 workflow_handler.go:116] Created a PipelineRun: my-devops-projectdq8jj/simple-pipeline-vkvcz

 I0118 02:26:07.726954       1 workflow_handler.go:116] Created a PipelineRun: my-devops-projectdq8jj/simple-pipeline-fxpcv
```

TODOs:
- [x] https://github.com/kubesphere/ks-jenkins/pull/77
- [x] https://github.com/kubesphere-sigs/ks-devops-helm-chart/pull/53

### Does this PR introduce a user-facing change??

```release-note
Add a webhook to process pipeline event
```

/label tide/merge-method-squash
/cc @kubesphere/sig-devops
